casper/src/main/scala/coop/rchain/casper/BlockRandomSeed.scala

```diff
@@ -8,21 +12,32 @@ import scodec.{Codec, TransformSyntax}
 final case class BlockRandomSeed(
     shardId: String,
     blockNumber: Long,
-    sender: PublicKey,
     validatorPublicKey: PublicKey,
     preStateHash: Blake2b256Hash
-)
+) {
+  def generateRandomNumber: Blake2b512Random =
+    Blake2b512Random(encode(this))
```

Comment:
Let's make default constructor private and add constructor on companion object. In that way we can add validation for parameters like shardId to be only ASCII characters.